api-client/src/endpoints/hooks/useAllPossibleSearchResults.ts

```diff
@@ -0,0 +1,98 @@
+import { useCallback, useMemo } from 'react'
+
+import createCategoriesEndpoint from 'api-client/src/endpoints/createCategoriesEndpoint'
+import createEventsEndpoint from 'api-client/src/endpoints/createEventsEndpoint'
+import useLoadAsync from 'api-client/src/endpoints/hooks/useLoadAsync'
+import useLoadFromEndpoint from 'api-client/src/endpoints/hooks/useLoadFromEndpoint'
+import loadSprungbrettJobs from 'api-client/src/endpoints/loadSprungbrettJobs'
+
+type useSearchParams = {
+  city: string
+  language: string
+  cmsApiBaseUrl: string
+}
+
+export type SearchResult = {
+  title: string
+  id: string | number
+  thumbnail?: string
+} & (
+  | {
+      content: string
+      path: string
+      location?: string
+      url?: string
+    }
+  | {
+      location: string
+      url: string
+      content?: string
+      path?: string
+    }
```

Comment:
I think we could just simplify this since we always use `path ?? url` and `location ?? content`, so we could also just do and store location in content and url in path.
```suggestion
      content: string
      path: string
    }
```